libcob/numeric.c

```diff
@@ -3543,7 +3543,12 @@ cob_add_int (cob_field *f, const int n, const int opt)
 	}
 #endif
 	/* n is single digit value added to positive field */
-	if (n > -10
+	if (
+#if 0 /* CHECKME: this does not work for negative numbers */
+	    n > -10
+#else
+	    n > 0
+#endif
```

Comment:
where / why doesn't that work? It gets the last stored number, then checks for negative => carry = -1, then operates on the positions one by one to the left; that approach sounds correct.

Reply:
I should have been more specific.
It breaks the test `DISPLAY: ADD and SUBTRACT w/o SIZE ERROR` (`data_display.at`).
Specifically this part:
```
       77  FLD0919A                       PIC S9(02)                         VALUE -06.
       77  FLD0919B                       PIC S9(01)                         VALUE +0.
...
       78  XPC-FLD0919B           VALUE 'v'.
...
           ADD FLD0919A        TO FLD0919B.
...
           IF FLD0919B (1:)    NOT EQUAL XPC-FLD0919B
              DISPLAY 'FLD0919B ==> ' FLD0919B
              DISPLAY 'EXPECTED ==> ' XPC-FLD0919B.
```
It outputs:
```
FLD0919B ==> -1
EXPECTED ==> v
```
So it "computes" -1 instead of -6 (because it forces the output to -1 when `carry == -1`).